pkg/querier/tripperware/shard_by.go

```diff
@@ -54,6 +56,12 @@ func (s shardBy) Do(ctx context.Context, r Request) (Response, error) {
 		level.Warn(logger).Log("msg", "error analyzing query", "q", r.GetQuery(), "err", err)
 	}
 
+	stats.AddExtraFields(
+		"shard_by.is_shardable", analysis.IsShardable(),
```

Comment:
I think the name `shard_by` is not very good because we can shard not only by in the future... But our middleware is called `shardBy` so this is probably ok

Reply:
Yeah.. i had a very hard time naming this middleware..

I'm open to suggestions and can create a follow up PR.